app/uk/gov/hmrc/agentsubscriptionfrontend/models/Tasks.scala

```diff
@@ -39,29 +65,47 @@ final case class AmlsTask(isMaa: Boolean, amlsData: Option[AmlsData]) extends Ta
   override val link: String = routes.AMLSController.showAmlsRegisteredPage().url
 }
 
-final case class MappingTask(
+final case class ContactDetailsEmailSubTask(isComplete: Boolean, showLink: Boolean) extends SubTask {
+  override val taskKey: String = "contactDetailsEmailSubTask"
+  override val link: String = ""
+}
+
+final case class ContactTradingNameSubTask(contactDetailsBusinessNameData: Option[String], showLink: Boolean)
+    extends SubTask {
+  override val taskKey: String = "contactDetailsTradingNameSubTask"
+  override val isComplete: Boolean = contactDetailsBusinessNameData.isDefined
+  override val link: String = ""
+}
+
+final case class ContactTradingAddressSubTask(
+  contactDetailsTradingAddressData: Option[BusinessAddress],
+  showLink: Boolean)
+    extends SubTask {
+  override val taskKey: String = "contactDetailsTradingAddressSubTask"
+  override val isComplete: Boolean = contactDetailsTradingAddressData.isDefined
+  override val link: String = ""
```

Comment:
you are overriding `link` with "" in all the subTasks, instead you can do this assignment in the base class and not override in sub classes ...
then it will be just little verbose